astropy/constants/__init__.py

```diff
@@ -50,8 +51,7 @@
     __doc__ += '\n'.join(_lines)
 
 
-# TODO: Re-implement in a way that is more consistent with astropy.units.
-#       See https://github.com/astropy/astropy/pull/7008 discussions.
+@deprecated('4.0', alternative='Use ScienceState for constants')
```

Comment:
Maybe more explicit: "use 'astropy.physical_constants' and 'astropy.astronomical_constants'"